### PR TITLE
Local variables used to eliminate compile time

### PR DESCRIPTION
**Pull request by Georgehe4**
#611 
@lukechurch @devoncarew 
Send a compile request on load for the embed views & save it locally for repeat requests. This means users don't experience the round trip latency of communicating with the server.

### CODE REVIEW

File: lib/embed/embed.dart [hunk 1]
```diff
@@ -56,9 +56,10 @@ class PlaygroundMobile {
   PaperIconButton _resetButton;
   PaperTabs _tabs;
 
-  Gist backupGist;
+  Map _lastRun;
```
Comment (lukechurch) [inline lib/embed/embed.dart]: Can we type this more specifically? Create an enum of the sources and have this as a Map<Enum, String>?

File: lib/embed/embed.dart [hunk 2]
```diff
@@ -103,6 +104,7 @@ class PlaygroundMobile {
       String id = url.queryParameters['id'];
       if (isLegalGistId(id)) {
         _showGist(id);
+        _backupRun();
```
Comment (lukechurch) [inline lib/embed/embed.dart]: Not really a backup, more a 'storePreviousResults'?

File: lib/embed/embed.dart [hunk 3]
```diff
@@ -533,11 +536,34 @@ class PlaygroundMobile {
...
+    if (_requestStored()) {
+      new Future(() {
+        CompileResponse response = _storedResponse;
+        if (executionService != null) {
+          return executionService.execute(
+              _context.htmlSource, _context.cssSource, response.result);
+        }
+      }).catchError((e) {
+        _showOutput('Error compiling to JavaScript:\n${e}', error: true);
+        _showError('Error compiling to JavaScript', '${e}');
```
Comment (lukechurch) [inline lib/embed/embed.dart]: Is it expected that _runbutton will be disabled if this path gets executed? That seems unfortunate as it means that in the presence of a server error the user is not going to get the chance to request another compile

File: lib/embed/embed.dart [hunk 4]
```diff
@@ -56,9 +56,10 @@ class PlaygroundMobile {
   PaperIconButton _resetButton;
   PaperTabs _tabs;
 
-  Gist backupGist;
+  Map _lastRun;
   Router _router;
-
+  
+  CompileResponse _storedResponse;
```
Comment (lukechurch) [inline lib/embed/embed.dart]: _storedCompile?

File: lib/embed/embed.dart [hunk 5]
```diff
@@ -554,13 +580,28 @@ class PlaygroundMobile {
     });
   }
 
+  bool _requestStored() {
```
Comment (lukechurch) [inline lib/embed/embed.dart]: _hasStoredRequest?

File: lib/embed/embed.dart [hunk 6]
```diff
@@ -533,11 +536,34 @@ class PlaygroundMobile {
       _editProgress.hidden(false);
     }
 
+    if (_requestStored()) {
+      new Future(() {
```
Comment (lukechurch) [inline lib/embed/embed.dart]: Does this future object need to be returned to anyone?

**Comment by Georgehe4** [comment]
@devoncarew @lukechurch  Ready for review again.

File: lib/embed/embed.dart [hunk 7]
```diff
@@ -46,6 +46,10 @@ void init() {
   _playground = new PlaygroundMobile();
 }
 
+enum FileType {
```
Comment (lukechurch) [inline lib/embed/embed.dart]: Either make it private or move it to somewhere shared?

File: lib/embed/embed.dart (hunk 3 above)
Comment (lukechurch) [inline lib/embed/embed.dart]: ping. This comment wasn't addressed.

File: lib/embed/embed.dart [hunk 8]
```diff
@@ -554,13 +583,28 @@ class PlaygroundMobile {
     });
   }
 
+  bool _hasStoredRequest() {
```
Comment (lukechurch) [inline lib/embed/embed.dart]: This should be a getter

File: lib/embed/embed.dart [hunk 9]
```diff
@@ -625,7 +669,14 @@ class PlaygroundMobile {
       e.text = description == null ? '' : description;
     }
   }
-
+  
+  void _setBackup () {
```
Comment (lukechurch) [inline lib/embed/embed.dart]: See previous comment - this isn't really a backup

**Comment by lukechurch** [comment]
lgtm, thanks!

**Comment by Georgehe4** [comment]
Merging.

**Pull request merged**